packages/middleware-ssec/src/index.ts

```diff
@@ -36,19 +36,25 @@ export function ssecMiddleware(options: PreviouslyResolved): InitializeMiddlewar
       for (const prop of properties) {
         const value: SourceData | undefined = (input as any)[prop.target];
         if (value) {
-          const valueView: Uint8Array = ArrayBuffer.isView(value)
-            ? new Uint8Array(value.buffer, value.byteOffset, value.byteLength)
-            : typeof value === "string"
-            ? options.utf8Decoder(value)
-            : new Uint8Array(value);
-          const encoded = options.base64Encoder(valueView);
+          let valueForHash: Uint8Array;
+          if (typeof value === "string") {
+            const isBase64Encoded = /^(?:[A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$/.test(value);
+            if (isBase64Encoded) {
+              valueForHash = new Uint8Array(Buffer.from(value, "base64"));
```

Comment:
Buffer.from isn't compatible with browsers

Reply:
Because that value is already base64 encoded, and does not need to be re-set on that target. 

Otherwise it would be 
```js
if (isBase64Encoded) {
              valueForHash = base64ToUint8Array(value);
              input[prop.target] = value; // and value is already set to be the target on line 37
}
```